packages/providers/src/transaction-summary/operations.ts

```diff
@@ -311,34 +320,74 @@ export function getContractCallOperations({
 }
 
 /** @hidden */
-export function getTransferOperations({ inputs, outputs }: InputOutputParam): Operation[] {
+export function getTransferOperations({
+  inputs,
+  outputs,
+  receipts,
+}: GetTransferOperationsParams): Operation[] {
   const coinOutputs = getOutputsCoin(outputs);
 
+  const [transferReceipt] = getReceiptsByType<TransactionResultTransferReceipt>(
+    receipts,
+    ReceiptType.Transfer
+  );
+
   let operations: Operation[] = [];
-  coinOutputs.forEach((output) => {
-    const input = getInputFromAssetId(inputs, output.assetId);
 
-    if (input) {
-      const inputAddress = getInputAccountAddress(input);
-      operations = addOperation(operations, {
-        name: OperationName.transfer,
-        from: {
-          type: AddressType.account,
-          address: inputAddress,
-        },
-        to: {
-          type: AddressType.account,
-          address: output.to.toString(),
-        },
-        assetsSent: [
-          {
-            assetId: output.assetId.toString(),
-            amount: output.amount,
+  // Possible transfer to contract
+  if (transferReceipt) {
+    const changeOutputs = getOutputsChange(outputs);
+    changeOutputs.forEach((output) => {
+      const { assetId } = output;
+      const [contractInput] = getInputsContract(inputs);
+      const utxo = getInputFromAssetId(inputs, assetId);
+
+      if (utxo && contractInput) {
+        const inputAddress = getInputAccountAddress(utxo);
+        operations = addOperation(operations, {
+          name: OperationName.transfer,
+          from: {
+            type: AddressType.account,
+            address: inputAddress,
           },
-        ],
-      });
-    }
-  });
+          to: {
+            type: AddressType.contract,
+            address: contractInput.contractID,
+          },
+          assetsSent: [
+            {
+              assetId: assetId.toString(),
+              amount: transferReceipt.amount,
+            },
+          ],
+        });
+      }
+    });
+  } else {
+    coinOutputs.forEach((output) => {
+      const input = getInputFromAssetId(inputs, output.assetId);
+      if (input) {
+        const inputAddress = getInputAccountAddress(input);
+        operations = addOperation(operations, {
```

Comment:
Is it ok to override the `operations` variable on each iteration of this `forEach` loop?

I might be totally out of context, but if only one execution reaches line `371`, maybe doing a `find` before the `forEach` (or breaking the loop when it matches the desired criteria) could be better.

Reply:
Couldn't there be more than one input in this context?

Reply:
@arboleya @digorithm good point. 

It is unclear to me if we only expect one operation at this part, or if multiple 'Transfer Asset' operations can happen on the same transaction.

Either way, this logic can be improved.

I've created this issue [here](https://github.com/FuelLabs/fuels-ts/issues/1411) to investigate any potential issues and to improve the code logic being used here